llpc/builder/llpcBuilderImplImage.cpp

```diff
@@ -1185,6 +1185,1185 @@ Value* BuilderImplImage::CreateImageStore(
+
+        // Linear interpolate
+        pResult = BilinearBlend(pAlpha, pBeta, pTL, pTR, pBL, pBR);
+    }
+
+    return pResult;
+}
+
+// =====================================================================================================================
+// Create YCbCr image sample internal
+Value* BuilderImplImage::YCbCrCreateImageSampleInternal(
+    SmallVectorImpl<Value*>& coords,     // ST coords
+    YCbCrSampleInfo*         pYCbCrInfo) // [In] YCbCr smaple information
+{
+    Value* pCoords = CreateInsertElement(Constant::getNullValue(VectorType::get(coords[0]->getType(), 2)),
+                                         coords[0],
+                                         uint64_t(0));
+
+    pCoords = CreateInsertElement(pCoords, coords[1], uint64_t(1));
+
+    return CreateImageSampleGather(pYCbCrInfo->pResultTy,
+                                   pYCbCrInfo->dim,
+                                   pYCbCrInfo->flags,
+                                   pCoords,
+                                   pYCbCrInfo->pImageDesc,
+                                   pYCbCrInfo->pSamplerDesc,
+                                   pYCbCrInfo->address,
+                                   pYCbCrInfo->instName,
+                                   pYCbCrInfo->isSample);
+}
+
+// =====================================================================================================================
+// Replace [beginBit, beginBit + adjustBits) bits with data in specific word
+Value* BuilderImplImage::ReplaceBitsInWord(
+    Value*      pWord,      // [in] Target wowrd
+    uint32_t    beginBit,   // The first bit to be replaced
+    uint32_t    adjustBits, // The number of bits should be replaced
+    Value*      pData)      // [in] The data used to replace specific bits
+{
+    uint32_t mask = ((1 << adjustBits) - 1) << beginBit;
+
+    Value* pMask = getInt32(mask);
+    Value* pInvMask = getInt32(~mask);
+    Value* pBeginBit = getInt32(beginBit);
+
+    if (auto pWordVecTy = dyn_cast<VectorType>(pWord->getType()))
+    {
+        if (isa<VectorType>(pData->getType()) == false)
+        {
+            pData = CreateVectorSplat(pWordVecTy->getNumElements(), pData);
+        }
+        pMask = CreateVectorSplat(pWordVecTy->getNumElements(), pMask);
+        pInvMask = CreateVectorSplat(pWordVecTy->getNumElements(), pInvMask);
+        pBeginBit = CreateVectorSplat(pWordVecTy->getNumElements(), pBeginBit);
+    }
+    else
+    {
+        if (auto pDataVecTy = dyn_cast<VectorType>(pData->getType()))
+        {
+            pWord = CreateVectorSplat(pDataVecTy->getNumElements(), pWord);
+            pMask = CreateVectorSplat(pDataVecTy->getNumElements(), pMask);
+            pInvMask = CreateVectorSplat(pDataVecTy->getNumElements(), pInvMask);
+            pBeginBit = CreateVectorSplat(pDataVecTy->getNumElements(), pBeginBit);
+        }
+    }
+
+    Value* pNewWord = CreateAnd(pWord, pInvMask); // (pWord & ~mask)
```

Comment:
This comment is excellent. Can we have something like for the previous function as well?